Board.java

```diff
@@ -76,62 +81,44 @@ public boolean isGameOver() {
         return hasWinner() || checkForDraw();
     }
 
-    public ArrayList<String> allWinningCombinations() {
-        ArrayList<String> combinations = new ArrayList<>();
+    public ArrayList<List<String>> allWinningCombinations() {
```

Comment:
You should return interfaces and not classes when possible, i.e. `List<List<String>>` instead of `ArrayList<List<String>>`. The only time you'd return `ArrayList` is if the caller depends on a method that's only on `ArrayList` but not on the `List` interface.